### PR TITLE
(#104) Make an option to rewrite JPeek's target directory if exists

### PR DESCRIPTION
**Pull request by llorllale**
as per #104 

Note:
I moved the `Files.exist(target)` check from `App.analyze()` to `Main.run()` because I didn't want to pollute `App` with this miscellaneous parameter and just wanted to let `App.analyze()` to focus on doing the *analysis*.

### CODE REVIEW

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this pull request

**Comment by 0crat** [comment]
Job [#152](https://github.com/yegor256/jpeek/issues/152) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#152](https://github.com/yegor256/jpeek/issues/152) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

File: src/main/java/org/jpeek/Main.java [hunk 1]
```diff
@@ -91,11 +102,50 @@ public static void main(final String... args) throws IOException {
      * @throws IOException If fails
      */
     private void run() throws IOException {
+        if (this.target.exists()) {
```
Comment (amihaiemil) [inline src/main/java/org/jpeek/Main.java]: @llorllale Well, this works, but it is very procedural. I think this logic should be inside an object called ``Target`` and it should be executed a few lines bellow, when calling ``this.target.toPath()``.

On the other hand, it's not your fault, the architecture headed towards this code from the beginning (``target`` was always a simple File, so all the logic about it is scattered around), so please, leave a puzzle for implementing such an object and refactoring this code :)

**Comment by llorllale** [comment]
@amihaiemil done, please check

**Comment by amihaiemil** [comment]
@rultor good to merge

**Comment by amihaiemil** [comment]
@yegor256 ping, don't forget about this one

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
Order was successfully finished: +15 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), total is [+330](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://datum.zerocracy.com/pages/policy.html#28): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+9315](https://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
The job [#152](https://github.com/yegor256/jpeek/issues/152) is now out of scope

**Comment by llorllale** [comment]
@yegor256 can we close this?

**Comment by yegor256** [comment]
@llorllale it's already closed :)

**Comment by llorllale** [comment]
Sorry, meant the bug

**Pull request merged**